### PR TITLE
add more log debug log for diagnose the version detect issue

### PR DESCRIPTION
**Pull request by missedone**
try to be helpful in diagnosing issue like https://github.com/cbeust/testng-eclipse/issues/338

### CODE REVIEW

File: remote/src/main/java/org/testng/remote/RemoteTestNG.java [hunk 1]
```diff
@@ -184,24 +195,30 @@ private static Version parseVersionFromManifest() throws Exception {
...
         }
 
-        if ("org.testng.TestNG".equals(mainAttrs.getValue("Main-Class"))) {
-          return toVersion(mainAttrs.getValue("Implementation-Version"));
+        if (ver == null && "org.testng.TestNG".equals(mainAttrs.getValue("Main-Class"))) {
+          ver = toVersion(mainAttrs.getValue("Implementation-Version"));
+        }
+
+        if (ver != null) {
+          p("parsed TestNG version at " + url);
```
Comment (missedone) [inline remote/src/main/java/org/testng/remote/RemoteTestNG.java]: print in which MANIFEST.MF detects the TestNG version

File: remote/src/main/java/org/testng/remote/RemoteTestNG.java [hunk 2]
```diff
@@ -155,11 +164,13 @@ private static Version parseVersionFromPom() throws Exception {
...
           "META-INF/maven/org.testng/testng/pom.properties");
       while (resources.hasMoreElements()) {
+        URL url = resources.nextElement();
         Properties props = new Properties();
-        try (InputStream in = resources.nextElement().openStream()) {
+        try (InputStream in = url.openStream()) {
           props.load(in);
         }
 
+        p("parsing TestNG version at " + url);
```
Comment (missedone) [inline remote/src/main/java/org/testng/remote/RemoteTestNG.java]: print in which pom.properties detects the TestNG version

File: remote/src/main/java/org/testng/remote/RemoteTestNG.java [hunk 3]
```diff
@@ -132,6 +135,12 @@ private static Version getTestNGVersion() {
     private static String getVersionFromClass() throws Exception {
       @SuppressWarnings("rawtypes")
       Class clazz = Class.forName("org.testng.internal.Version");
+      if (isDebug()) {
+        CodeSource codeSource = clazz.getProtectionDomain().getCodeSource();
+        if (codeSource != null) {
+          p("loaded class " + clazz.getCanonicalName() + " at " + codeSource.getLocation());
```
Comment (missedone) [inline remote/src/main/java/org/testng/remote/RemoteTestNG.java]: print in which class detects the TestNG version

**Comment by juherr** [comment]
LGTM

**Comment by juherr** [comment]
The coveralls-maven-plugin is failing on java9 due to https://github.com/trautonen/coveralls-maven-plugin/issues/112

**Pull request merged**